File: C_Sum_Of_Cubes.cpp
#include <cmath>
#include <iostream>

#include <unordered_set>

using namespace std;

int main() {
    int t;
    cin >> t;
    while (t--) {
        unsigned long long int n;
        cin >> n;
        unordered_set<unsigned long long int> seen;
        for (unsigned long long i = 1; i < ceil(cbrt(n)); i++)
            seen.insert(pow(i, 3));
        bool flag = false;
        for (auto num : seen) {
            if (seen.find(n - num) != seen.end()) {
                cout << "YES\n";
                flag = true;
                break;
            }
        }
        if (not flag)
            cout << "NO\n";
    }
}

File: C_Sum_Of_Cubes.py
import math
from random import randint

t = int(input())
for _ in range(t):

    n = int(input())
    x = randint(1, 1_000_000)
    run = 0
    seen = set()
    for i in range(1, math.ceil(math.cbrt(n))):
        seen.add((i**3) ^ x)
    for num in seen:
        if (n - (num ^ x)) ^ x in seen:
            print("YES")
            break
    else:
        print("NO")
